File: backend/StockEase/src/main/java/edu/cit/batucan/StockEase/feature/auth/dto/AuthResponse.java
package edu.cit.batucan.StockEase.feature.auth.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AuthResponse {
    
    private UserDto user;
    private String accessToken;
    private String refreshToken;
}

File: mobile/app/src/main/java/com/it342/stockease/feature/auth/model/AuthResponse.kt
package com.it342.stockease.feature.auth.model

data class AuthResponse(
    val success: Boolean,
    val data: AuthData?,
    val error: ErrorData?,
    val timestamp: Any?
)

data class AuthData(
    val user: UserData?,
    val accessToken: String?,
    val refreshToken: String?
)

data class UserData(
    val email: String,
    val firstname: String,
    val lastname: String,
    val role: String
)

data class ErrorData(
    val code: String?,
    val message: String?
)
